From dfc415852d33b9765dad754ac185e039516ee0b6 Mon Sep 17 00:00:00 2001
From: Daniel Wagner <wagi@monom.org>
Date: Mon, 20 Jun 2011 19:24:57 +0200
Subject: [PATCH] device: Do not return error for unhandled device

If device_probe returns an error for a device where no
driver has been registered, the device will be destroyed.
This happens when for example the wpa_supplicant is
started after ConnMan. In this case we don't have any
wifi device registered at the core when wpa_supplicant starts.
---
 src/device.c |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/src/device.c b/src/device.c
index 067c794..3e3e96d 100644
--- a/src/device.c
+++ b/src/device.c
@@ -1199,7 +1199,7 @@ static int device_probe(struct connman_device *device)
 	}
 
 	if (device->driver == NULL)
-		return -ENODEV;
+		return 0;
 
 	return setup_device(device);
 }
-- 
1.7.5.2

